### ISSUE TITLE
Can't get the paper results for Book-crossing dataset

### ISSUE THREAD

**Issue by Gtarget** [hwwang55/RippleNet#4]
Hi, I just run the code using Booking-crossing dataset for CTR prediction task without modifying any line of code, but it seems I could not reproduce the paper results, see my result:
![image](https://user-images.githubusercontent.com/18683363/46794230-0053db80-cd7a-11e8-86a1-243d43bfe778.png)

I don't understand why. Could you kindly examine the default hyperparameter settings and help me reproduce the paper results? 
Thanks a lot.

**Comment by hwwang55** [comment]
I tried on my computer just now and the code seems working well. Here is the result:

![image](https://user-images.githubusercontent.com/28680408/46809012-37d57e80-cda0-11e8-8641-9160a3d98550.png)

I found that the code can hardly reproduce the reported result on Windows. You can try it on Linux. Another thing is did you use the default setting for book-crossing (see main.py)? You cannot just run "python main.py --dataset book". The default setting is for movielens.

**Comment by hwwang55** [comment]
I tried on my computer just now and the code seems working well. Here is the result:

![default](https://user-images.githubusercontent.com/28680408/46809532-3c4e6700-cda1-11e8-99d4-3cf9db117fd7.JPG)

I found that the code can hardly reproduce the reported result on Windows. You can try it on Linux. Another thing is did you use the default setting for book-crossing (see main.py)? You cannot just run "python main.py --dataset book". The default setting is for movielens.

**Comment by hwwang55** [comment]
That table is from the ealier version of the paper. We have found mistakes when pre-processing the book-crossing dataset and re-run the experiments for RippleNet and all baselines. The correct result can be found in the latest version of this paper in arxiv and the procedding of cikm (may not be available yet).

**Comment by Gtarget** [comment]
Oh,there. I'll go check the latest version, thx.

**Comment by ZJUWeifeng** [comment]
Hi, I have run the code under the default settings for the Book-Crossing dataset at least 10 times and my result is similar to Gtarget's. 
![screenshot from 2018-10-18 15-06-18](https://user-images.githubusercontent.com/24896538/47136931-c2116b80-d2e7-11e8-84fe-50bacfe48a1f.png)
As you can see, the best auc is 0.6951, which is worse than Wide&Deep's auc.
Could you tell me how to achieve the best auc reported in your paper? Are there any differences between the knowledge graph you use and the one you provide here?

**Comment by hwwang55** [comment]
I ran the code for 5 times just now, and here are the results:

![1](https://user-images.githubusercontent.com/28680408/47146054-da8c8080-d2fd-11e8-8919-e2f660ee2825.JPG)

![2](https://user-images.githubusercontent.com/28680408/47146060-db251700-d2fd-11e8-9508-4cda9d7cc54a.JPG)

![3](https://user-images.githubusercontent.com/28680408/47146065-dceeda80-d2fd-11e8-9ff0-4a06d9fd7214.JPG)

![4](https://user-images.githubusercontent.com/28680408/47146069-deb89e00-d2fd-11e8-8339-045395e498ba.JPG)

![5](https://user-images.githubusercontent.com/28680408/47146075-e1b38e80-d2fd-11e8-8eab-a7c0b044e1a0.JPG)

There is indeed a chance that the model get stuck in AUC of 0.7, but the probability is small. I have checked the parameters and there is no problem. The only thing different is that I anonymized the IDs of entities (i.e., re-indexing) in the released datasets according to the privacy policy of Microsoft, but that should have no effect on the performance. I'll check that later.

**Comment by ZJUWeifeng** [comment]
Thanks a lot. I will try it again.

**Comment by ubear** [comment]
> Thanks a lot. I will try it again.

how about the result?

**Comment by ZJUWeifeng** [comment]
> > Thanks a lot. I will try it again.
> 
> how about the result?

By using different random seeds, I get better results, which is 0.72~0.73 on the Book-Crossing dataset. Some random seeds will get stuck, and some will not.

**Comment by yorkchu1995** [comment]
> By using different random seeds, I get better results, which is 0.72~0.73 on the Book-Crossing dataset. Some random seeds will get stuck, and some will not.

The seed is？

**Comment by zhenchengchang** [comment]
> I tried on my computer just now and the code seems working well. Here is the result:
> 
> I found that the code can hardly reproduce the reported result on Windows. You can try it on Linux. Another thing is did you use the default setting for book-crossing (see main.py)? You cannot just run "python main.py --dataset book". The default setting is for movielens.

i want to ask why the results in your picture decrease after better results.

**Comment by cmaxhao** [comment]
I think it may suffer from overfitting.
